lib/livebook_web/live/home_live.ex

```diff
@@ -320,6 +320,21 @@ defmodule LivebookWeb.HomeLive do
     {:noreply, socket}
   end
 
+  def handle_info({:set_file, file}, socket) do
+    socket =
+      case import_notebook(file) do
+        {:ok, {notebook, messages}} ->
+          socket
+          |> put_import_warnings(messages)
+          |> create_session(notebook: notebook, file: file, origin: {:file, file})
+
+        {:error, error} ->
+          put_flash(socket, :error, Livebook.Utils.upcase_first(error))
+      end
```

Comment:
We do the same on `"open"` event, let's move this to `open_notebook(socket, file)`.